### PR TITLE
Vignettes as tutorials

### PR DESCRIPTION
**Pull request by LienReyserhove**
## Description

This PR contains the vignettes as tutorials, as described in #262 
All vignettes can be found in the `tutorials/content/tutorials` folder. 
One tutorial equals one vignette. The name of the file follows the convention `vignette` + `package_name`+ `vignette_name`.
The structure of the `index.md` uses the following structure:

``` markdown
---
title: "..."
description: "Vignette for the R package ...."
authors: ...
date: ...
categories: ["r"]
tags: ["r", "vignette", ...]
output: 
    md_document:
        preserve_yaml: true
        variant: markdown_github
---

See the vignette/tutorial at <link>
```
@florisvdh @peterdesmet @ThierryO the revision is all yours

## Task list

- [x] My tutorial or article is placed in a subfolder of `tutorials/content`
- [x] The filename of my tutorial or article is `index.md`. In case of an Rmarkdown tutorial I have knitted my `index.Rmd` to `index.md` (both files are pushed to the repo). 
- [x] I have included `tags` in the YAML header (see the tags listed in the [tutorials website side bar](https://inbo.github.io/tutorials/) for tags that have been used before)
- [x] I have added `categories` to the YAML header and my category tags are from the [list of category tags](https://github.com/inbo/tutorials/blob/master/static/list_of_categories)

## Previewing the pull request

Thanks to GitHub Actions, an artifact (=zip file) of the rendered website is automatically created for each pull request.

### CODE REVIEW

**Review by ElsLommelen — CHANGES_REQUESTED** [review]
Don't forget to add PR #294 first.

**Comment by ThierryO** [comment]
Instead of linked to the individual vignettes, we could link to https://inbo.r-universe.dev/ui#articles instead.

**Comment by ElsLommelen** [comment]
@ThierryO When linking to inbo.r-universe, to what extend can we find these manuals when using the tags?  For instance, I would like to query the inboveg database using R and I don't know how to do this.  So I search on the tutorials website how to do it.  At the moment there is still an old copy of the inbodb package vignette on inboveg that explains this, and that pops up directly when searching on the subject of databases, but if this is deleted, how will a visitor of the tutorials site be aware of the fact that there is a package with specific functions to solve his problem?

Maybe it is not necessary to have such link to all vignettes of all packages (e.g. to me it seems less relevant for specific non-coding-related subjects such as as package dhcurve), and maybe links to vignettes on related subjects can be grouped on one site (e.g. put all vignettes on querying INBO databases together, or all vignettes that have the same combination of tags), but for subjects that are typically searched for on the tutorials website, to me it seems useful for tutorials site visitors to at least learn that an INBO package on this subject exists.  (If I don't have experience with coding on a specific topic, e.g. query databases or manipulate raster geodata, I would search on the tutorials site, rather than checking if INBO has a package on it for any chance.)

On the other hand I agree with @ThierryO that it may not be necessary to link to every single vignette and give all vignette a separate site, which will certainly become a disaster for maintenance.  In inbodb, probably 2 new vignettes will be added this year, so it seems easier for maintenance to just refer to [inbo.github.io/inbodb](https://inbo.github.io/inbodb/) where vignettes for different databases can be found (I plan on giving the vignettes more visibility on this main page).  Probably a similar exercise can be done for other vignettes and packages?

So I think it is certainly helpful to have these tutorials that refer to package( vignette)s, but maybe try to refer to packages rather than vignettes (if the subject is related), and try to group different links on one page as much as possible.

**Comment by peterdesmet** [comment]
The goals of this PR was to elevate the work we collectively put in vignettes and make them easily accessible as tutorials via the Tutorials website. As @ElsLommelen explains, it allows to search for a topic and find vignettes as well as regular tutorials on that subject, which wouldn't work if we just refer to r-universe.

Regarding maintenance, I don't think that's going to be too bad, especially if we only do it once a year (@LienReyserhove correct me if I'm wrong). As @ElsLommelen suggests, we could bundle Vignettes for _some_ packages on a single page, but that also comes with the cognitive burden (during maintenance) of having to decide which ones to bundle, what authors, topics, etc. to apply, ... with little added benefit.

So, until this becomes unwieldy, I like the approach of having one Vignette = one Tutorial.

**Comment by ElsLommelen** [comment]
> we could bundle Vignettes for _some_ packages on a single page, but that also comes with the cognitive burden (during maintenance) of having to decide which ones to bundle, what authors, topics, etc. to apply, ... with little added benefit.

One advantage for the website visitor to bundle vignettes is to avoid having long lists after searching for a topic (tag), each of it referring to a separate page with just one link.

Easiest would be to bundle topics that have got the same tags, but indeed: authors.  Maybe mention only @LienReyserhove as an author because she has composed the pages?  The package or vignette authors are mentioned at the linked sites, I suppose?

Anyway, I leave it up to the maintainer(s) to decide if this small advantage outweighs the extra effort to bundle vignettes.

File: content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md
Comment (florisvdh) [inline content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md:4]: You need to make sure that for every author there’s a `AUTHOR_ID.toml` file under `data/authors` and the match with that ID must be exact; see https://inbo.github.io/tutorials/create_tutorial/. The above is just an example; there are several cases where this is still a problem. Once a commit is added that solves this, the GH action workflow that builds the website will hopefully succeed so that you can preview the result. Currently this failed.

**Review by florisvdh — COMMENTED** [review]
Thank you for undertaking this effort @LienReyserhove.

Considering the general strategy, I can live with current approach, but indeed it will need maintenance and it requires that you (or a colleague) is willing to update this as needed or at least yearly, as suggested above.

Shouldn't we include the package name as a title prefix to make it visually easier to recognize the package vignettes in current approach? See idea in #262.

Example titles:

- `inborutils`: GBIF scientific name matching
- `inborutils`: Check the coordinate system
- ...

I think this will make the fact of being a vignette, as well as the relationships between vignettes of the same package, more prominent in the webpages which enlist tutorials - more so than with the added description which is visually less prominent. The description could then be left as is, or even be generalized to a fixed description 'R package vignette'.

Another reason for adding the title prefix is because various vignette titles are to be understood within the context of the package. Without having the prefix or without reading the current description in combination with the title, several titles cannot be understood well. Example of such a title: _Define the date period to download_.

File: content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md
Comment (peterdesmet) [reply content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md:4]: I will fix this

Comment (peterdesmet) [reply content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md:4]: Is this case sensitive?

Comment (peterdesmet) [reply content/tutorials/vignette_n2khab_v025_geospatial_hab/index.md]: Apparently it is. And horrible to debug. 🤪

**Comment by peterdesmet** [comment]
@florisvdh I cannot use markdown in titles, so it will have to be:

> inborutils: GBIF scientific name matching

Not

> `inborutils`: GBIF scientific name matching

Here's a comparison between the two approach (as part of description vs part of title):

Screenshot 2022-01-27 at 13 20 10

Do you still prefer to prepend titles?

**Comment by florisvdh** [comment]
Thanks for solving the author IDs!

Sure, I still would prepend the titles with the package  name.

What about:

image

This is the UTF-8 'package' character (https://www.utf8icons.com/character/128230/package).

See 792a016a for the syntaxis used, if you like this (mind the spaces).

**Comment by peterdesmet** [comment]
I like the emoji approach, but would maybe use it on the description, so it doesn't become to visually overloading in the overview of titles. Compare: 
Screenshot 2022-01-27 at 15 22 51

Makes it also easier to copy/paste without errors.

**Comment by florisvdh** [comment]
Two more variations:

image
(a1aadbdb)

image
(4b6d48cc)

Corresponding tutorial pages of the three variations:

image


image


image

**Comment by peterdesmet** [comment]
All issue and feedback resolved, thanks for the input! A huge thanks to @LienReyserhove 👏👏👏  for hunting down all the vignettes, translating these into tutorials, and weathering the feedback and format changes. 😄 @LienReyserhove I have also updated the spreadsheet to reflect the current state.

**Comment by florisvdh** [comment]
Just spotted a dwelling 📦:

image

File: content/tutorials/vignette_inlatools_rwprior/index.md
Comment (florisvdh) [inline content/tutorials/vignette_inlatools_rwprior/index.md:2]:
```suggestion
title: "inlatools: Getting a grasp on the random walk hyperparameter"
```

File: content/tutorials/vignette_rgbif_downloads/index.md
Comment (florisvdh) [inline content/tutorials/vignette_rgbif_downloads/index.md:3]:
```suggestion
title: "rgbif: Downloading data from GBIF"
description: "📦 R package vignette."
```

**Review by florisvdh — APPROVED** [review]
Nice work @LienReyserhove and @peterdesmet, thanks :tada:. Just a few things left as far as I could tell!

**Review by peterdesmet — APPROVED** [review]

**Comment by peterdesmet** [comment]
@ElsLommelen no need for a thorough review (we all had a check already, especially @florisvdh), but your approval is needed before this PR can be merged. 😄

**Review by ElsLommelen — APPROVED** [review]
Sorry, I didn't know that I had to approve to remove this 'request changes'.  Here you are!  (I trust @florisvdh did it thoroughly, no time to do it myself today.)

**Pull request merged**
